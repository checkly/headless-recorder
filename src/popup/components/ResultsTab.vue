<template>
  <div class="tab results-tab">
    <div class="tabs">
      <button v-for="tab in tabs" :key="tab" class="tabs__action" v-bind:class="{'selected': activeTab === tab}" @click.prevent="changeTab(tab)">
        <span v-if="tab === 'playwright'">ðŸŽ­</span>
        <img v-if="tab === 'puppeteer'" src="/images/puppeteer.png" width="16" />
        <span class="tabs__action--text">{{ tab }}</span>
      </button>
    </div>

    <div class="content">
      <pre v-if="code()" v-highlightjs="code()">
        <code class="javascript"></code>
      </pre>
      <pre v-else>
        <code>No code yet...</code>
      </pre>
    </div>
  </div>
</template>
<script>
export const TYPE = {
  PUPPETEER: 'puppeteer',
  PLAYWRIGHT: 'playwright'
}

export default {
  name: 'ResultsTab',
  props: {
    puppeteer: {
      type: String, 
      default: ''
    },
    playwright: {
      type: String, 
      default: ''
    },
    options: {
      type: Object,
      default: () => ({})
    }
  },
  data () {
    return {
      activeTab: TYPE.PUPPETEER,
      tabs: [TYPE.PUPPETEER, TYPE.PLAYWRIGHT]
    }
  },
  mounted() {
    if (this.options && this.options.code && this.options.code.showPlaywrightFirst) {
      this.activeTab = TYPE.PLAYWRIGHT
      this.tabs = this.tabs.reverse()
    }
    this.$emit('update:tab', this.activeTab)
  },
  methods: {
    code() {
      return this.activeTab === TYPE.PUPPETEER ? this.puppeteer : this.playwright
    },
    changeTab(tab) {
      this.activeTab = tab
      this.$emit('update:tab', tab)
    }
  }
  }
</script>
<style lang="scss" scoped>
  @import "~styles/_variables.scss";

  .results-tab {
    .content {
      display:flex;
      flex-direction:column;
      height:100%;
      .generated-code {
        flex: 1;
        height:100%;
        overflow: auto;
        display: flex;
        flex-direction: column-reverse;
        max-height: $max-content-height;
      }

      pre {
        padding: 0 $spacer;
        font-size: 12px;
      }
      .code {
        font-family: Consolas, Monaco, monospace;
        padding: $spacer;
      }
    }
  }

  .tabs {
    display: flex;
    border-bottom: 1px solid $gray-lighter;
    &__action {
      padding: 12px 20px;
      border: 0;
      background: transparent;
      cursor: pointer;
      font-size: 14px;
      color: $gray-dark;
      outline: none;
      border-bottom: 4px solid transparent;
      text-transform: capitalize;
      display: flex;
      align-items: center;
      &--text {
        margin-left: 10px;
      }
      &.selected {
        border-bottom-color: $blue;
      }
    }
  }
</style>
