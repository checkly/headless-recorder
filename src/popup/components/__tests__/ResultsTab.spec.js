import { createLocalVue, mount } from '@vue/test-utils'
import ResultsTab from '../ResultsTab'

describe('RecordingTab.vue', () => {
  let localVue
  beforeEach(() => {
    localVue = createLocalVue()
    localVue.directive('highlightjs', () => {})
  })

  test('it has the correct pristine / empty state', () => {
    const wrapper = mount(ResultsTab, { localVue })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('code.javascript').exists()).toBe(false)
  })

  test('it show a code box when there is code', () => {
    const wrapper = mount(ResultsTab, { localVue })
    wrapper.setProps({ puppeteer: `await page.click('.class')` })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('code.javascript').exists()).toBe(true)
  })

  test('it render tabs for puppeteer & playwright', () => {
    const wrapper = mount(ResultsTab, { localVue })
    expect(wrapper.findAll('.tabs__action').length).toEqual(2)
  })

  test('it render playwright first when option is present', async () => {
    const wrapper = await mount(ResultsTab, {
      localVue,
      propsData: {
        options: {
          code: {
            showPlaywrightFirst: true
          }
        }
      }
    })
    expect(wrapper.find('.tabs__action').text()).toEqual('ðŸŽ­  playwright')
  })
})
