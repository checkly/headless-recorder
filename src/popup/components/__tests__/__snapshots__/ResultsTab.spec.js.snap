// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingTab.vue it has the correct pristine / empty state 1`] = `
<div
  class="tab results-tab"
>
  <div
    class="tabs"
  >
    <button
      class="tabs__action selected"
    >
      <!---->
       
      <img
        src="/images/puppeteer.png"
        width="16"
      />
       
      <span
        class="tabs__action--text"
      >
        puppeteer
      </span>
    </button>
    <button
      class="tabs__action"
    >
      <span>
        ðŸŽ­
      </span>
       
      <!---->
       
      <span
        class="tabs__action--text"
      >
        playwright
      </span>
    </button>
  </div>
   
  <div
    class="content"
  >
    <pre>
            
      <code>
        No code yet...
      </code>
      
    
    </pre>
  </div>
</div>
`;

exports[`RecordingTab.vue it show a code box when there is code 1`] = `
<div
  class="tab results-tab"
>
  <div
    class="tabs"
  >
    <button
      class="tabs__action selected"
    >
      <!---->
       
      <img
        src="/images/puppeteer.png"
        width="16"
      />
       
      <span
        class="tabs__action--text"
      >
        puppeteer
      </span>
    </button>
    <button
      class="tabs__action"
    >
      <span>
        ðŸŽ­
      </span>
       
      <!---->
       
      <span
        class="tabs__action--text"
      >
        playwright
      </span>
    </button>
  </div>
   
  <div
    class="content"
  >
    <pre>
            
      <code
        class="javascript"
      />
      
    
    </pre>
  </div>
</div>
`;
